### DISCUSSION TITLE
How do you generate devcontainer-collection.json?

### DISCUSSION THREAD

**Discussion by SiegfredRodriguez** [devcontainers/spec#513] [Q&A] [answered]
TLDR; How do you generate devcontainer-collection.json?

I attempted creating my own template collection, however i get an error.
`[21 ms] (!) ERR: Failed to fetch devcontainer-collection.json`

As per documentation, devcontainer-collection.json is supposed to be auto generated,
but there seems to be no documentation on how or when this is done.

I cloned the templates project, modified it and executed this command :
`devcontainer templates publish -n SiegfredRodriguez/jdk-postgres -r g
hcr.io .`

**Comment by joshspicer** [comment, marked as answer]
Looking at your repo here where you run the GitHub action, I'd suggest first fixing the [issue here](https://github.com/SiegfredRodriguez/jdk-postgres/actions/runs/11574921181/job/32220277808#step:3:21) and trying to run the tool again.  

You can fix this by renaming  `src/jdk-postgres/.devcontainer/.devcontainer.json` to `src/jdk-postgres/.devcontainer/devcontainer.json` (without the leading dot on `devcontainer.json`).  I think that will solve your issue


That [specific error is a bit misleading](https://github.com/devcontainers/cli/blob/9ba1fdaa11dee087b142d33e4ac13c5788392e34/src/spec-node/templatesCLI/publish.ts#L70), thank you for pointing that out

**Reply by SiegfredRodriguez** [reply]
Thank you very much, the error message was confusing.
Putting the dot before the json was muscle memory.

**Reply by joshspicer** [reply]
I've removed that error message in https://github.com/devcontainers/cli/pull/925 - thanks again!
